CDP4WspDatabaseAuthentication/Cdp4WspDatabaseAuthenticatorConnector.cs

```diff
@@ -0,0 +1,111 @@
+    public class Cdp4WspDatabaseAuthenticatorConnector : AuthenticatorConnector<AuthenticatorWspProperties>
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="Cdp4WspDatabaseAuthenticatorConnector"/> class.
+        /// </summary>
+        /// <param name="properties">
+        /// The properties.
+        /// </param>
+        public Cdp4WspDatabaseAuthenticatorConnector(AuthenticatorWspProperties properties)
+            : base(properties)
+        {
+        }
+
+        /// <summary>
+        /// Gets a value indicating whether the connector is up.
+        /// </summary>
+        public override bool IsUp
+        {
+            get
+            {
+                this.IsDown = false;
+                return true;
+            }
+        }
+
+        /// <summary>
+        /// Gets a value indicating whether the connector is down.
+        /// </summary>
+        public override bool IsDown { get; set; }
+
+        /// <summary>
+        /// Gets or sets the status message of the <see cref="IAuthenticatorConnector"/>.
+        /// </summary>
+        public override string StatusMessage { get; set; }
+
+        /// <summary>
+        /// Gets the <see cref="IAuthenticatorConnector"/> name for display purposes.
+        /// </summary>
+        public override string Name
+        {
+            get { return "CDP4 Wsp Database Authenticator"; }
+        }
+
+        /// <summary>
+        /// Authenticate the <see cref="AuthenticationPerson"/> information against the supplied password.
+        /// </summary>
+        /// <param name="person">
+        /// The <see cref="AuthenticationPerson"/> information to authenticate.
+        /// </param>
+        /// <param name="password">
+        /// The password to authenticate against.
+        /// </param>
+        /// <returns>
+        /// True if the <see cref="AuthenticationPerson"/> could be authenticated.
+        /// </returns>
+        public override bool Authenticate(AuthenticationPerson person, string password)
+        {
+            return this.ValidatePassword(password, person);
+        }
+
+        /// <summary>
+        /// Verifies that the password that the login has provided is correct.
+        /// </summary>
+        /// <param name="password">
+        /// The input password to test.
+        /// </param>
+        /// <param name="person">
+        /// The <see cref="AuthenticationPerson"/> that this password should authenticate against.
+        /// </param>
+        /// <returns>
+        /// True if the passwords match.
+        /// </returns>
+        private bool ValidatePassword(string password, AuthenticationPerson person)
+        {
+            var result = false;
+
+            var serverSalts = (this.Properties as IAuthenticatorWspProperties)?.ServerSalts;
+
+            if (serverSalts == null)
+            {
+                return false;
+            }
+
+            foreach (var serverSalt in serverSalts)
```

Comment:
there are multiple server salts?

Reply:
Indeed there are multiple server salts because there can be multiple wsp source servers, and when the `Cdp4WspDatabaseAuthenticatorConnector` will try to validate it will try with each salt.

Reply:
@samatrhea this was done with presumption that maybe in the future you could indeed migrate models from multiple servers into one. In the current flow of things the array will only contain at most one salt indeed.

Reply:
ok, understood